src/game/core/stage/Stage.ts

```diff
@@ -0,0 +1,70 @@
+import { Entity, Resources } from 'game/core';
+
+export default class Stage {
+  private _context: CanvasRenderingContext2D;
+  private _resources: Resources;
+  private _backgroundPattern?: CanvasPattern | null;
+  private _entities: Record<string, Entity[] | []> | Record<string, never>;
+
+  constructor(context: CanvasRenderingContext2D, resources: Resources) {
+    this._context = context;
+    this._resources = resources;
+    this._entities = {};
+  }
+
+  addEntitiesToKey(key: string, entities: Entity[]): void {
+    if (!this._entities[key]) {
+      this._entities[key] = [];
+    }
+    this._entities[key].push(...entities);
```

Comment:
❗️
Для большей читабельности стоит везде отбивать условия от остальных кусков кода
```suggestion
    if (!this._entities[key]) {
      this._entities[key] = [];
    }
    
    this._entities[key].push(...entities);
```